src/frameobject.ts

```diff
@@ -141,6 +143,36 @@ class Py_FrameObject implements IPy_FrameObj {
             }
         }
     }
+
+    emptyStack() {
+        while (this.stack.length > 0) {
+            this.pop();
+        }
+    }
+    raise_exception_here(t: Thread, message: string, type: string): void {
+        var val: IPy_Object = (<any> builtins)[type];
+        t.exc = val;
+        this.push(val);
+        t.addToTraceback(message);
+        this.tryCatchException(t);
+    }
+
+    tryCatchException(t: Thread): boolean {
+        // Whenever an exception occurs, tries to find a handler and if it can't outputs the traceback 
+        // search in current frame
+        if (!find_exception_handler(this)) {
+            // Search for exception handler in previous frames
+            if (!find_in_prev(this,t))
+            {
+                // no exception handler found, write traceback and exit the thread
+                t.throwException();
+                return false;
+            }
+        }
+        // save line number on which exception occurred
+        t.raise_lno = (this.codeObj.firstlineno) + addr2line(this);
```

Comment:
I'm not sure why this is stored on the thread object. Is this for traceback construction?